src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,84 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.StandardOpenOption;
+
 public class WorkWithFile {
+
+    private static final int TYPE_INDEX = 0;
+    private static final int VALUE_INDEX = 1;
+    private static final String RESULT_RECORD = "result";
+    private static final String SUPPLY_TYPE = "supply";
+    private static final String COMMA_SEPARATOR = ",";
+    private static final String BUY_TYPE = "buy";
+    private static final String EMPTY_STRING = "";
+
     public void getStatistic(String fromFileName, String toFileName) {
+        String[] records = readFromFile(fromFileName);
+        String[] convertedRecords = recordsConverter(records);
+        File report = createOrReplace(toFileName);
+        writeRecordsToFile(convertedRecords, report);
+    }
+
+    public String[] readFromFile(String fileName) {
+        String dataFromFile = EMPTY_STRING;
+        StringBuilder readerStringBuilder;
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(fileName))) {
+            readerStringBuilder = new StringBuilder();
+            dataFromFile = bufferedReader.readLine();
+            while (dataFromFile != null) {
+                readerStringBuilder.append(dataFromFile).append(System.lineSeparator());
+                dataFromFile = bufferedReader.readLine();
+            }
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read the file", e);
+        }
+        return readerStringBuilder.toString().split(System.lineSeparator());
```

Comment:
You can try to use List instead of builder 
```suggestion
                result.add(dataFromFile);
                dataFromFile = bufferedReader.readLine();
            }
        } catch (IOException e) {
            throw new RuntimeException("Can't read the file", e);
        }
        return result.toArray();
```